herddb-core/src/main/java/herddb/security/UserManager.java

```diff
@@ -28,4 +31,14 @@
 public abstract class UserManager {
 
     public abstract String getExpectedPassword(String username);
+
+    public void authenticate(String username, char[] pwd) throws IOException {
+        String password = new String(pwd);
+        String expectedPassword = getExpectedPassword(username);
+        if (expectedPassword == null
+                || password == null
```

Comment:
@eolivelli this seems unaddressed by your last commit. 
We can simply remove the null check for `password`. If `pwd` is `null` (idk if this is possible) we get an npe, otherwise the checks on the expected pwd are enough

Reply:
@eolivelli this seems unaddressed by your last commit. 
We can simply remove the null check for `password`. If `pwd` is `null` (idk if this is possible) we get an npe, otherwise the checks on the expected pwd are enough

Reply:
I missed it.
Patch updated.
thanks for the heads up @dmercuriali